bloom-housing/bloom
# feat: open modal when selecting paper app download language

# Pull Request Template

## Issue Overview

This PR addresses #3634


## Description

This PR changes the behavior of the `Download application` button on a public listing. Now, a modal opens with a radio selection of the available languages. If only one paper application is available, it just downloads that language instead of opening the modal. The design is in the ticket.

## How Can This Be Tested/Reviewed?

Ensure that if a listing has only one paper application added the download application link does not open a modal and just downloads that paper app.

Ensure that if a listing has multiple paper applications uploaded that a modal pops with the options of each language, and that hitting the confirm button downloads the respective app.

After chatting with design, we settled on not having a success toast due to the browser's native alert about downloaded files.

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
